internal/test/e2e/run.go

```diff
@@ -482,3 +470,47 @@ func logTestGroups(logger logr.Logger, instancesConf []instanceRunConf) {
 	}
 	logger.V(1).Info("Running tests in parallel", "testsGroups", testGroups)
 }
+
+func getHardwarePool(storageBucket string) ([]*api.Hardware, error) {
+	awsSession, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("creating aws session for tests: %v", err)
+	}
+	err = s3.DownloadToDisk(awsSession, os.Getenv(tinkerbellHardwareS3FileKeyEnvVar), storageBucket, e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download tinkerbell hardware csv: %v", err)
+	}
+
+	hardware, err := api.NewHardwareSliceFromFile(e2eHardwareCsvFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Tinkerbell hardware: %v", err)
+	}
+	return hardware, nil
+}
+
+func reserveTinkHw(conf instanceRunConf, invQueue *hwQueue) {
+	now := time.Now()
+	after := now.Add(hwPollingTimeout)
+	for {
+		if now.After(after) {
+			conf.logger.V(1).Info("TestRunner timed out waiting for hardware", "instanceId", conf.instanceId)
+			break
+		}
+		invQueue.mu.Lock()
```

Comment:
The mutex locking and unlocking should ideally happen at the queue side. The tests should only care about getting the requested hardware from the queue, not handling concurrency for the queue.